skiko/src/commonMain/kotlin/org/jetbrains/skiko/Events.kt

```diff
@@ -1,11 +1,97 @@
 package org.jetbrains.skiko
 
 // Clicked mouse buttons bitmask.
-object SkikoMouseButtons {
-    const val NONE = 0
-    const val LEFT = 1 shl 0
-    const val RIGHT = 1 shl 1
-    const val MIDDLE = 1 shl 2
+class SkikoMouseButtons(val value: Int) {
+    companion object {
+        val NONE = SkikoMouseButtons(0)
+        val LEFT = SkikoMouseButtons(1)
+        val RIGHT = SkikoMouseButtons(2)
+        val MIDDLE = SkikoMouseButtons(4)
+        val BUTTON_1 = SkikoMouseButtons(1)
+        val BUTTON_2 = SkikoMouseButtons(2)
+        val BUTTON_3 = SkikoMouseButtons(4)
+        val BUTTON_4 = SkikoMouseButtons(8)
+        val BUTTON_5 = SkikoMouseButtons(16)
+        val BUTTON_6 = SkikoMouseButtons(32)
+        val BUTTON_7 = SkikoMouseButtons(64)
+        val BUTTON_8 = SkikoMouseButtons(128)
+    }
+
+    fun has(value: SkikoMouseButtons): Boolean {
+        if (value.value and this.value != 0) {
+            return true
+        }
+        return false
+    }
+
+    override fun toString(): String {
+        return mutableListOf<String>().apply {
+            if (has(SkikoMouseButtons.LEFT)) {
+                add("LEFT")
+            }
+            if (has(SkikoMouseButtons.RIGHT)) {
+                add("RIGHT")
+            }
+            if (has(SkikoMouseButtons.MIDDLE)) {
+                add("MIDDLE")
+            }
+            if (has(SkikoMouseButtons.BUTTON_4)) {
+                add("BUTTON_4")
+            }
+            if (has(SkikoMouseButtons.BUTTON_5)) {
+                add("BUTTON_5")
+            }
+            if (has(SkikoMouseButtons.BUTTON_6)) {
+                add("BUTTON_6")
+            }
+            if (has(SkikoMouseButtons.BUTTON_7)) {
+                add("BUTTON_7")
+            }
+            if (has(SkikoMouseButtons.BUTTON_8)) {
+                add("BUTTON_8")
+            }
+            if (isEmpty()) {
+                add("NONE")
+            }
+        }.toString()
+    }
+}
+
+class SkikoInputModifiers(val value: Int) {
+    companion object {
+        val EMPTY = SkikoInputModifiers(0)
+        val META = SkikoInputModifiers(1)
+        val CONTROL = SkikoInputModifiers(2)
+        val ALT = SkikoInputModifiers(4)
+        val SHIFT = SkikoInputModifiers(8)
+    }
+
+    fun has(value: SkikoInputModifiers): Boolean {
+        if (value.value and this.value != 0) {
+            return true
+        }
+        return false
+    }
+
+    override fun toString(): String {
+        return mutableListOf<String>().apply {
+            if (has(SkikoInputModifiers.META)) {
+                add("META")
+            }
+            if (has(SkikoInputModifiers.CONTROL)) {
+                add("CONTROL")
+            }
+            if (has(SkikoInputModifiers.ALT)) {
+                add("ALT")
+            }
+            if (has(SkikoInputModifiers.SHIFT)) {
+                add("SHIFT")
+            }
+            if (isEmpty()) {
```

Comment:
Seems wrong, at least for me gives `PointerEvent: [EMPTY] + DOWN + [LEFT]`

Reply:
I've refactored [toString] method to be cleaner.